src/components/TextEditor.tsx

```diff
@@ -1,62 +1,75 @@
 "use client";
 import React, { useRef, useState } from "react";
-import Editor from "@monaco-editor/react";
+import MonacoEditor, { OnMount } from "@monaco-editor/react";
+import { editor } from "monaco-editor";
+import { useParams } from "next/navigation";
 
-export default function TextEditor() {
-  const editorRef = useRef(null);
+interface TextEditorProps {
+  instrument: string;
+  specification: string;
+  setSpecification: (specification: string) => void;
+}
+
+async function getSpecification(instrument: string) {
+  const response = await fetch(`/api/instrument/${instrument}/specification`);
+
+  if (!response.ok) {
+    throw new Error("Failed to fetch data");
+  }
+
+  return JSON.stringify(await response.json());
+}
+
+export default function TextEditor(props: TextEditorProps) {
+  // [slug] from the app route is converted to the [instrument] in the api route
+  const { slug } = useParams();
+
+  const editorRef = useRef<editor.IStandaloneCodeEditor | null>(null);
   const [scheduledChange, setScheduledChange] = useState(false);
   const options = {
     automaticLayout: true,
     formatOnPaste: true,
     lineWrap: "on",
```

Comment:
I am slightly confused by the `lineWrap`  option, does this do anything? Because I believe the type of of the options is here https://microsoft.github.io/monaco-editor/typedoc/interfaces/editor.IStandaloneEditorConstructionOptions.html and I don't see that option.

Reply:
No it looks like it doesn't (I thought it was affecting whether a horizontal scroll bar was visible, but that is being controlled by wordWrap). 

I can't find the original source which suggested lineWrap as an option. It will be removed.